rust/kernel/clk.rs

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Common clock framework.
+//!
+//! C header: [`include/linux/clk.h`](../../../../include/linux/clk.h)
+
+use crate::{
+    bindings, device,
+    error::{from_kernel_err_ptr, Error, Result},
+    str::CStr,
+};
+
+/// Represents `struct clk *`.
+///
+/// # Invariants
+///
+/// The pointer is valid.
+pub struct Clk(*mut bindings::clk);
+
+impl Clk {
+    /// Returns value of rate field of `struct clk`.
+    pub fn get_rate(&self) -> usize {
+        // SAFETY: the pointer is valid by the type invariant.
+        unsafe { bindings::clk_get_rate((*self).0) as usize }
+    }
+
+    /// Do not use in atomic context.
+    pub fn prepare_enable(&mut self) -> Result {
+        // SAFETY: the pointer is valid by the type invariant.
+        let ret = unsafe { bindings::clk_prepare_enable((*self).0) };
+        if ret < 0 {
+            return Err(Error::from_kernel_errno(ret));
+        }
+
+        Ok(())
+    }
+}
+
+/// Lookup and obtain a managed reference to a clock producer.
```

Comment:
nit: please use the third-person when describing something. IOW, "Looks up and obtains [...]"

Reply:
Sure, thanks.